Aufgabe: Pool A: Charakterklassen (#269)
PM-Dungeon/PM-Lecture: markdown/assignments/pool_a/charakterklassen/index.md

hidden: true
---

## Ziel

> In Rollenspielen ist es üblich, dass die Spielfigur einer bestimmen Klasse zugehörig ist. In dieser Aufgabe sollen Sie verschiedene Charakterklassen implementieren.

Reviewer: ```suggestion
In Rollenspielen ist es üblich, dass die Spielfigur einer bestimmen Spiel-Klasse zugehörig ist. In dieser Aufgabe sollen Sie verschiedene Charakterklassen implementieren.
```

## Voraussetzungen

Um diese Aufgabe zu lösen, sollten Sie am besten bereits ein Kampfsysteme (Nahkampf, Fernkampf, Magie) und Skills umgesetzt haben.

## Charakterklassen

Überlegen Sie sich zuerst, welche Charakterklassen Sie implementieren wollen. Sie sollen mindestens drei unterschiedliche Charakterklassen anbieten.
Überlegen UND notieren Sie dabei, was diese Charakterklassen gemeinsam haben und worin sie sich unterscheiden. Geben Sie jeder Charakterklasse eine passende Bezeichnung.

> Implementieren Sie nun die verschiedenen Charakterklassen im Sinne des OOP-Gedanken. Überlegen Sie auch, welche Pattern Sie ggf. verwenden können. Begründen Sie Ihre Entscheidungen schriftlich.

Reviewer: ```suggestion
Implementieren Sie nun die verschiedenen Charakterklassen im Sinne des OOP-Gedankens. Überlegen Sie auch, welche Pattern Sie ggf. verwenden können. Begründen Sie Ihre Entscheidungen schriftlich.
```

> Zu Beginn des Spiels soll der Spieler die Klasse Auswählen können (oder haben Sie eine bessere Idee?).

Reviewer: ```suggestion
Zu Beginn des Spiels soll der Spieler die Klasse auswählen können (oder haben Sie eine bessere Idee?).
```

hidden: true
---

## Ziel

> In Rollenspielen gibt die Charakterklasse an, welche Werte (Anzahl der Lebenspunkte, Kampfschaden, `[Manapunkte]({{<ref "https://de.wikipedia.org/wiki/Mana_(Spiele)" >}})`{=markdown} etc.) die Spielfigur hat, welche Waffen sie verwenden und welche Fähigkeiten sie erlernen kann. Typische Charakterklassen sind Krieger, Zauberer oder Schurke.

Reviewer: ```suggestion
In Rollenspielen gibt die Charakterklasse an, welche Werte (Anzahl der Lebenspunkte, Kampfschaden, [Mana-Punkte](https://de.wikipedia.org/wiki/Mana_(Spiele)) etc.) die Spielfigur hat, welche Waffen sie verwenden und welche Fähigkeiten sie erlernen kann. Typische Charakterklassen sind Krieger, Zauberer oder Schurke.
```
